File: data_structures/queue.js
//Create a queue using pseudoclassical instantiation
var Queue = function(limit) {
  this._storage = [];
  this._size = 0;
  this._limit = limit || 0;
}

Queue.prototype.enqueue = function(val) {
  //add item to beginning of queue and increment size
  if (this._size < this._limit) {
    this._size++;
    this._storage.unshift(val);
  } else {
    return "Cannot enqueue value, queue is full";
  }
}

Queue.prototype.dequeue = function() {
  //Remove and return val at end of stack if it is nonempty
  if (this._size > 0) {
    this._size--;
    return this._storage.pop();
  }
  return null;
}

Queue.prototype.size = function() {
  return this._size;
}

File: data_structures/Python/queue.py
#Create a queue class
class Queue():

	def __init__(self, limit=1000):
		self._storage = []
		self._limit = limit
		self._size = _size

	def enqueue(self, item):
		if self._size < self._limit:
			self._size += 1
			self._storage.insert(0,item)

	def dequeue(self):
		if self._size > 0:
			self._size -= 1
			return self._storage.pop()

	def size(self):
		return self._size()